tests/lib/lore_test.sh

```diff
@@ -421,4 +421,22 @@ function test_save_new_lore_config()
   assert_equals_helper 'Wrong lore.config contents' "$LINENO" "$expected" "$output"
 }
 
+function test_get_kernel_tree_branches()
+{
+  declare -A branches
+
+  # Mock for command 'git branch --verbose'
+  # shellcheck disable=SC2317
+  function git()
+  {
+    printf ' branch1  \t cafebabe commit_subject1\n  branch2 \t\t deadbeef commit_subject2\n\t\t branch3   c00010ff commit_subject3'
+  }
+  export -f git
+
+  get_kernel_tree_branches "$kernel_tree_path" 'branches'
```

Comment:
Where this `$kernel_tree_branches` came from?
Also, I think that a test that creates a fake repository with multiple branches is more valuable for this case; in other words, I recommend you not use the `git` mock and use the function in a fake repository created by the test. I think we have some examples of that in the kw code.

Reply:
Hi @rodrigosiqueira 
That was my first approach as I agree that mocking the `git` in this superficial way is not ideal. The tests worked perfectly but I was met with this message when trying to the sample repository of the project:

<img width="977" alt="Screen Shot 2023-06-10 at 20 18 41" src="https://github.com/kworkflow/kworkflow/assets/95055784/b537f289-d107-4625-ba08-247b9f919913">

My approach was to create a sample git repository and copy it to the project's tree. From what I understand this warning means that a submodule (a git repo insider another) must be declared together with a remote. Is it okay to ignore this message or am I making it wrong?

Reply:
In the setup phase, you can create a temporary folder followed by `git init`; that way, you will have a valid git repository in a temporary folder. In the tear_down phase, you can delete the temporary folder.

Reply:
Thanks a lot! I'm sure that this approach will do 😄